src/main/java/uk/gov/hmcts/juror/api/moj/service/jurormanagement/JurorAppearanceServiceImpl.java

```diff
@@ -650,21 +658,19 @@ public void confirmJuryAttendance(UpdateAttendanceDto request) {
     }
 
     private void checkExistingAttendance(JurorNonAttendanceDto request, LocalDate nonAttendanceDate) {
-        // check if there is already an appearance record for the juror for the non-attendance date
+        final String locCode = request.getLocationCode();
         final String jurorNumber = request.getJurorNumber();
-        appearanceRepository.findByJurorNumberAndPoolNumberAndAttendanceDate(jurorNumber, request.getPoolNumber(),
-                nonAttendanceDate)
+        // check if there is already an appearance record for the juror for the non-attendance date
+        appearanceRepository.findByLocCodeAndJurorNumberAndAttendanceDate(locCode, jurorNumber, nonAttendanceDate)
```

Comment:
do we not need to take pool number into account as well?   guessing they will only be in one pool on a particular day..

Reply:
Appearances used to be based on juror_number and pool number but this was since changed to be juror_number and loc_code